### PR TITLE
Support gRPC sync grouped dynamic configurations.

### LINKED ISSUE #7081: Dynamic configuration for a set of configurations.
Currently, `ConfigChangeWatcher` only allows implementors to register a key for a dynamic configuration update. With more configurations being dynamic, we may not provide all keys, such as all MAL scripts.

@wankai123 @mrproliu @kezhenxu94 @dmsolr or other @apache/skywalking-committers do we have any ideas about how to add another dynamic configuration API for a set of data?

I don't think we have to keep compatible with the existing one, the keys about this feature are 
1. Watcher could indicate a name/key for multiple items.
2. Every item of the name/key could be changed/updated separately, which means it could be maintained separately.

### PR DESCRIPTION
**Pull request by wankai123**
### Support gRPC sync grouped dynamic configurations.
- [ ] If this is non-trivial feature, paste the links/URLs to the design doc.
- [X] Update the documentation to include this new feature.
- [X] Tests(including UT, IT, E2E) are added to verify the new feature.
- [ ] If it's UI related, attach the screenshots below.

- [X] If this pull request closes/resolves/fixes an existing issue, replace the issue number. Closes #7081.
- [X] Update the [`CHANGES` log](https://github.com/apache/skywalking/blob/master/CHANGES.md).

### CODE REVIEW

**Comment by kezhenxu94** [comment]
Please format the new files

File: oap-server/server-configuration/grpc-configuration-sync/src/main/test/java/org/apache/skywalking/oap/server/configuration/grpc/GRPCConfigurationTest.java [hunk 1]
```diff
@@ -0,0 +1,177 @@
+/*
```
Comment (kezhenxu94) [inline oap-server/server-configuration/grpc-configuration-sync/src/main/test/java/org/apache/skywalking/oap/server/configuration/grpc/GRPCConfigurationTest.java:1]: Please check, the test resource directory structure is wrong, how do you test this locally?

oap-server/server-configuration/grpc-configuration-sync/src/main/test/java/org/apache/skywalking/oap/server/configuration/grpc/GRPCConfigurationTest.java

Comment (kezhenxu94) [reply oap-server/server-configuration/grpc-configuration-sync/src/main/test/java/org/apache/skywalking/oap/server/configuration/grpc/GRPCConfigurationTest.java:1]: src/main/test is typically wrong

File: oap-server/server-configuration/grpc-configuration-sync/src/main/java/org/apache/skywalking/oap/server/configuration/grpc/GRPCConfigWatcherRegister.java [hunk 2]
```diff
@@ -70,14 +70,42 @@ public GRPCConfigWatcherRegister(RemoteEndpointSettings settings) {
...
+                        groupName);
+                    groupConfigTable.addGroupConfigItems(groupConfigItems);
+                    rspGroupConfigItems.getItemsList().forEach(item -> {
+                        groupConfigItems.add(new ConfigTable.ConfigItem(item.getName(), item.getValue()));
+                    });
+                }
+            });
+            this.groupUuid = responseUuid;
+        } catch (Exception e) {
+            log.error("Remote config center [" + settings + "] is not available.", e);
```
Comment (kezhenxu94) [inline oap-server/server-configuration/grpc-configuration-sync/src/main/java/org/apache/skywalking/oap/server/configuration/grpc/GRPCConfigWatcherRegister.java:107]: Do not use string concatenation, use `{}`

File: oap-server/server-configuration/grpc-configuration-sync/src/main/test/java/org/apache/skywalking/oap/server/configuration/grpc/GRPCConfigurationTest.java [hunk 3]
```diff
@@ -0,0 +1,177 @@
...
+    private ConfigChangeWatcher singleWatcher;
+    private GroupConfigChangeWatcher groupWatcher;
+    /**
+     * 0:init, 1:change, 2:no change, 3:delete
+     */
+    public static volatile int singleDataFlag;
+    /**
+     * 0:init, 1:change, 2:no change, 3:delete
+     */
+    public static volatile int groupDataFlag;
```
Comment (kezhenxu94) [inline oap-server/server-configuration/grpc-configuration-sync/src/main/test/java/org/apache/skywalking/oap/server/configuration/grpc/GRPCConfigurationTest.java:50]: Using a static variable to pass states between classes is not a good practice, although it's only a test, please consider other mechanism, e.g. passing an `AtomicInteger` as a constructor parameter of `MockGRPCConfigService` class

File: oap-server/server-configuration/grpc-configuration-sync/src/main/test/java/org/apache/skywalking/oap/server/configuration/grpc/GRPCConfigurationTest.java [hunk 4]
```diff
@@ -0,0 +1,188 @@
...
+            v = groupWatcher.groupItems().get("item2")) {
+        }
+        assertEquals("2000", groupWatcher.groupItems().get("item2"));
+        //no change
+        groupDataFlag = 2;
+        TimeUnit.SECONDS.sleep(3);
+        for (String v = groupWatcher.groupItems().get("item1");
+            v.equals("100");
+            v = groupWatcher.groupItems().get("item2")) {
+        }
```
Comment (kezhenxu94) [inline oap-server/server-configuration/grpc-configuration-sync/src/main/test/java/org/apache/skywalking/oap/server/configuration/grpc/GRPCConfigurationTest.java:130]:
```suggestion
        for (String v = groupWatcher.groupItems().get("item1");
            v.equals("100");
            v = groupWatcher.groupItems().get("item1")) {
        }
```

File: oap-server/server-configuration/grpc-configuration-sync/src/main/test/java/org/apache/skywalking/oap/server/configuration/grpc/GRPCConfigurationTest.java (hunk 1 above)
Comment (wankai123) [reply oap-server/server-configuration/grpc-configuration-sync/src/main/test/java/org/apache/skywalking/oap/server/configuration/grpc/GRPCConfigurationTest.java]: :sweat:

**Comment by wu-sheng** [comment]
Please fix the UT.

File: oap-server/server-configuration/grpc-configuration-sync/src/main/java/org/apache/skywalking/oap/server/configuration/grpc/GRPCConfigWatcherRegister.java [hunk 5]
```diff
@@ -70,14 +70,42 @@ public GRPCConfigWatcherRegister(RemoteEndpointSettings settings) {
...
+                        groupName);
+                    groupConfigTable.addGroupConfigItems(groupConfigItems);
+                    rspGroupConfigItems.getItemsList().forEach(item -> {
+                        groupConfigItems.add(new ConfigTable.ConfigItem(item.getName(), item.getValue()));
+                    });
+                }
+            });
+            this.groupUuid = responseUuid;
+        } catch (Exception e) {
+            log.error("Remote config center [{}}] is not available.", settings, e);
```
Comment (kezhenxu94) [inline oap-server/server-configuration/grpc-configuration-sync/src/main/java/org/apache/skywalking/oap/server/configuration/grpc/GRPCConfigWatcherRegister.java:107]:
```suggestion
            log.error("Remote config center [{}] is not available.", settings, e);
```

File: oap-server/server-configuration/grpc-configuration-sync/src/test/java/org/apache/skywalking/oap/server/configuration/grpc/GRPCConfigurationTest.java [hunk 6]
```diff
@@ -0,0 +1,177 @@
...
+        ConfigurationServiceGrpc.ConfigurationServiceBlockingStub blockingStub = ConfigurationServiceGrpc.newBlockingStub(
+            grpcServerRule.getChannel());
+        Whitebox.setInternalState(register, "stub", blockingStub);
+        initWatcher();
+        assertNotNull(provider);
+    }
+
+    @Test(timeout = 20000)
+    public void shouldReadUpdated() throws Exception {
+        AtomicInteger dataFlage = new AtomicInteger(0);
```
Comment (kezhenxu94) [inline oap-server/server-configuration/grpc-configuration-sync/src/test/java/org/apache/skywalking/oap/server/configuration/grpc/GRPCConfigurationTest.java:66]:
```suggestion
        AtomicInteger dataFlag = new AtomicInteger(0);
```

File: oap-server/server-configuration/grpc-configuration-sync/src/test/java/org/apache/skywalking/oap/server/configuration/grpc/GRPCConfigurationTest.java [hunk 7]
```diff
@@ -0,0 +1,177 @@
...
+        dataFlage.set(3);
+        TimeUnit.SECONDS.sleep(1);
+        for (String v = singleWatcher.value(); v.equals("300"); v = singleWatcher.value()) {
+        }
+        assertEquals("", singleWatcher.value());
+    }
+
+    @Test(timeout = 20000)
+    public void shouldReadUpdated4Group() throws Exception {
+        AtomicInteger dataFlage = new AtomicInteger(0);
```
Comment (kezhenxu94) [inline oap-server/server-configuration/grpc-configuration-sync/src/test/java/org/apache/skywalking/oap/server/configuration/grpc/GRPCConfigurationTest.java:97]:
```suggestion
        AtomicInteger dataFlag = new AtomicInteger(0);
```

**Review by kezhenxu94 — DISMISSED** [review]
LGTM, some nits

**Review by kezhenxu94 — APPROVED** [review]

**Pull request merged**